uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/performance/DeviceMonitor.java

```diff
@@ -0,0 +1,219 @@
+        PerformanceProfileMonitor<?> monitor = profilers.get(performanceProfile);
+
+        if (monitor == null) {
+            throw new IllegalArgumentException("The performanceProfile does not have a monitor.");
+        }
+
+        loadProcessInfo();
+
+        return classOfT.cast(monitor.getStats(processInfo));
+    }
+
+    /**
+     * Fetch the current process information
+     *
+     * @return the current process information
+     */
+    public static ProcessInfo getProcessInfo() {
+        return processInfo;
+    }
+
+    /**
+     * Updates or adds a new profiling monitor for a particular profile
+     *
+     * @param profile  the profile to update
+     * @param profiler the profile monitor to use
+     * @param <T>      the type of the data that will be collected by the {@link PerformanceProfileMonitor}
+     */
+    public static <T> void setProfiler(PerformanceProfile profile, PerformanceProfileMonitor<T> profiler) {
+        profilers.put(profile, profiler);
+    }
+
+    /**
+     * Returns the current profile monitor tool that's being used by the profile
+     *
+     * @param profile {@link PerformanceProfile}
+     * @return the profile monitor tool
+     */
+    public static PerformanceProfileMonitor<?> getProfiler(PerformanceProfile profile) {
+        return profilers.get(profile);
+    }
+
+    /**
+     * Fetches the current process information by executing several adb shell commands.
+     */
+    private static void loadProcessInfo() {
+        processInfo = new ProcessInfo();
+
+        final int pid = android.os.Process.myPid();
+
+        String command = "top -b -n 1 -p " + pid;
+        String[] commandResult = AdbShellUtils.executeShellCommand(command).split("\n");
+
+        final String[] processInfoString = commandResult[5].trim().split("\\s+");
+        final String cpuUsage = processInfoString[8];
+        final String memoryUsage = processInfoString[9];
+
+        command = "cat /proc/meminfo";
+        commandResult = AdbShellUtils.executeShellCommand(command).split("\n");
+
+        final long memTotal = Long.parseLong(commandResult[0].trim().split("\\s+")[1]);
+        final long memFree = Long.parseLong(commandResult[1].trim().split("\\s+")[1]);
+        final long memAvailable = Long.parseLong(commandResult[2].trim().split("\\s+")[1]);
+
+        final int numCpuCores = Runtime.getRuntime().availableProcessors();
+
+        processInfo = ProcessInfo.builder()
+                .cpuUsage(Double.parseDouble(cpuUsage) / numCpuCores)
+                .freeSystemMemory(memFree)
+                .availableSystemMemory(memAvailable)
+                .totalSystemMemory(memTotal)
+                .usedSystemMemory(memTotal - memAvailable)
+                .memoryUsage(Double.parseDouble(memoryUsage))
+                .packageName(packageName)
+                .pid(pid)
+                .uid(getApplicationUid()).build();
+    }
```

Comment:
Looks like `ProcessInfo` is just instantiated here only...is there a reason for it to be declared as a static that isn't a constant?

Reply:
Makes sense, should be a constant.